### PR TITLE
Implement get inclusion proof by index

### PR DESCRIPTION
**Pull request by Martin2112**
Not directly part of CT api though can be used to implement get-entry-and-proof. Part of #93.

### CODE REVIEW

File: server/trillian_log_server.go [hunk 1]
```diff
@@ -89,8 +89,53 @@ func (t *TrillianLogServer) QueueLeaves(ctx context.Context, req *trillian.Queue
 }
 
 // GetInclusionProof obtains the proof of inclusion in the tree for a leaf that has been sequenced.
+// Similar to the get proof by hash handler but one less step as we don't need to look up the index
 func (t *TrillianLogServer) GetInclusionProof(ctx context.Context, req *trillian.GetInclusionProofRequest) (*trillian.GetInclusionProofResponse, error) {
-	return nil, ErrNotImplemented
+	// Reject obviously invalid tree sizes and leaf indices
```
Comment (AlCutter) [inline server/trillian_log_server.go]: Seems like there's a lot of common code between this and the `...ByHash` funcs, could it be factored out?

Comment (Martin2112) [inline server/trillian_log_server.go]: Possibly but I need to untangle pending branches first.

Comment (Martin2112) [inline server/trillian_log_server.go]: Most of the duplication is in the tests. Apart from parameter validation they call the same function to retrieve the proofs.

**Comment by Martin2112** [comment]
I've switched tests to use the parameterized version where appropriate. I think this is better now.

**Pull request merged**